src/main/kotlin/org/dashj/platform/dpp/util/Cbor.kt

```diff
@@ -289,50 +193,45 @@ object Cbor {
             is Boolean -> arrayBuilder.add(value)
             is Int -> arrayBuilder.add(value.toLong())
             is BigInteger -> arrayBuilder.add(value.toLong())
-            is Number -> arrayBuilder.add(value.toLong())
             is Float -> arrayBuilder.add(value)
             is Long -> arrayBuilder.add(value)
             is Double -> arrayBuilder.add(value)
             is ByteArray -> arrayBuilder.add(value)
             is Identifier -> arrayBuilder.add(value.toBuffer())
+            is Number -> arrayBuilder.add(value.toLong())
             null -> arrayBuilder.add(SimpleValue(SimpleValueType.NULL))
-            else -> arrayBuilder.add(value.toString()) // ?
+            else -> throw IllegalArgumentException("No converter for $value")
         }
     }
 
     private fun readJSONObject(obj: co.nstant.`in`.cbor.model.Map): MutableMap<String, Any?> {
         val resultMap = HashMap<String, Any?>()
-        val sortedKeys = ArrayList<DataItem>()
-        sortedKeys.addAll(obj.keys)
+        val keys = ArrayList<DataItem>()
+        keys.addAll(obj.keys)
```

Comment:
This must have been copy/pasted from writeJSONObject.

There is no reason to sort keys on a read operation.  The purpose of sorting keys on a write operation is to allow for comparison of CBOR byte data regardless of the Map implementation (trees or hashtables) that may order the keys in different ways,.